services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -713,6 +714,36 @@ public void routeOptionsWithUnrecognizedObjectsAndArrays() throws IOException {
     RouteOptions.fromJson(mutatedRouteOptionsJson.toString()).toUrl("testToken");
   }
 
+  @Test
+  public void routeOptionsDoNotAddAccessTokenFromJsonToUrl() {
+    Gson gson = new GsonBuilder().create();
+    JsonObject routeOptionsJson = gson.fromJson(optionsJson, JsonObject.class);
+    routeOptionsJson.addProperty("access_token", "wrong");
+    String routeOptionsJsonString = routeOptionsJson.toString();
+
+    URL url = RouteOptions.fromJson(routeOptionsJsonString).toUrl("right");
+
+    assertFalse(
+      "url shouldn't contain tokens from json: " + url,
+      url.getQuery().contains("access_token=wrong")
```

Comment:
```suggestion
      url.getQuery().contains("access_token")
```

Reply:
The URL should contain 1 `access_token`, the suggested check will fail the test.

Reply:
yeah, that makes sense. Lets' also check there's exact `1` access token

Reply:
added